### PR TITLE
Feature/networking client queries

### PR DESCRIPTION
**Pull request by vacla**
* Adds response packet
* Adds input packet
* Adds silent messages
* Fixes reload networking

### CODE REVIEW

**Review by DarkKronicle — APPROVED** [review]

**Comment by vacla** [comment]
@Intelli would it be possible to look at this before next Coreprotect release?

**Comment by Intelli** [comment]
Can you describe what these changes are doing?

What's the use case for silent messages?

**Comment by vacla** [comment]
@Intelli this is a follow up from the previous networking PR i made, i tagged you in discord with an example for what the input response and response packages are meant for, the silent flag is just so nothing will be sent in chat but it is totally up to the user if nothing will be sent in chat or not since there is an autopage also so the users won't need to go thru every page but that is also optional and wont happen if the user doesnt want it, i can provide you with a jar so you can try it out.

**Comment by Intelli** [comment]
Please provide the examples here.

Any functionality intended for client-side mods, that can be accomplished in client-side mods, should be left in those mods, rather than being implemented in the plugin.

**Comment by vacla** [comment]
@Intelli 

image

the data in the ui is partially provided with the handshake, the actions and the worlds since the client won't know about these directly, all the data if submitted happens over de input channel, this will happen as a format like a command that you would sent as usual, and the popup that you see is the response packet that was sent by coreprotect

**Comment by vacla** [comment]
There are also examples how to use the networking api in the documentation.

File: lang/en.yml [hunk 1]
```diff
@@ -106,6 +106,7 @@ LOGGING_ITEMS: "{0} items left to log. Please wait..."
 LOGGING_TIME_LIMIT: "Logging time limit reached. Discarding data and shutting down."
 LOOKUP_BLOCK: "{0} {placed|broke} {1}."
 LOOKUP_CONTAINER: "{0} {added|removed} {1} {2}."
+LOOKUP_FINISHED: "Lookup Finished"
```
Comment (Intelli) [inline lang/en.yml:109]: Please remove unused LOOKUP_FINISHED phrase

File: src/main/java/net/coreprotect/command/CommandHandler.java [hunk 2]
```diff
@@ -679,7 +681,7 @@ else if (material != null) {
                                     restricted.add(material);
                                 }
                                 else {
-                                    Chat.sendMessage(player, Color.DARK_AQUA + "CoreProtect " + Color.WHITE + "- " + Phrase.build(Phrase.INVALID_INCLUDE, argument));
+                                    Chat.sendResponse(player, Color.DARK_AQUA + "CoreProtect " + Color.WHITE + "- " + Phrase.build(Phrase.INVALID_INCLUDE, argument), "coreprotect:search", silent);
```
Comment (Intelli) [inline src/main/java/net/coreprotect/command/CommandHandler.java:684]: "coreprotect:search" is not a valid command (coreprotect:lookup)

File: src/main/java/net/coreprotect/listener/channel/PluginChannelInputListener.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    }
+                    catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+            Bukkit.getScheduler().runTask(CoreProtect.getInstance(), new sendLookupPages());
+        }
+        catch (Exception exception) {
+            Chat.console(exception.toString());
```
Comment (Intelli) [inline src/main/java/net/coreprotect/listener/channel/PluginChannelInputListener.java:95]: Duplicate exception information?

File: src/main/java/net/coreprotect/utility/Chat.java [hunk 4]
```diff
@@ -61,14 +63,46 @@ public static void sendGlobalMessage(CommandSender user, String string) {
...
+    public static void sendResponse(CommandSender user, String message, String type, boolean silent) {
+        sendMessageSilent(user, message, silent);
+        sendPluginChatResponseMessage(user, message, type);
+    }
+
+    public static void sendPluginChatResponseMessage(CommandSender user, String message, String type) {
+        PluginChannelResponseListener.getInstance().sendData(user, message, type);
+    }
+
+    public static void sendMessageSilent(CommandSender user, String message, boolean silent) {
```
Comment (Intelli) [inline src/main/java/net/coreprotect/utility/Chat.java:85]: `sendMessageSilent`, `sendComponentSilent`, and `sendComponentSilent` seem unnecessary.

Can just move the `sendMessageSilent` logic into `sendResponse` for improved clarity.

**Review by Intelli — CHANGES_REQUESTED** [review]

File: lang/en.yml (hunk 1 above)
Comment (vacla) [reply lang/en.yml:109]: It is being used here: https://github.com/PlayPro/CoreProtect/pull/211/files#diff-aeeaed8bd7931b9f87f7c7995b43968b15886536833939a5ab1a8ec32fbc8085R1101

Comment (Intelli) [reply lang/en.yml:109]: Link just takes me to changed files, but I only see 3 references:
`en`.yml`, `language.java`, and `Phrase.java`

File: src/main/java/net/coreprotect/utility/Chat.java (hunk 4 above)
Comment (vacla) [reply src/main/java/net/coreprotect/utility/Chat.java:85]: `sendMessageSilent`, `sendComponentSilent` and `sendComponentSilent` are all used in `LookupCommand`, these methods are all handled differently.

File: src/main/java/net/coreprotect/listener/channel/PluginChannelInputListener.java (hunk 3 above)
Comment (vacla) [reply src/main/java/net/coreprotect/listener/channel/PluginChannelInputListener.java]: Removed Chat.console method call

File: src/main/java/net/coreprotect/command/CommandHandler.java (hunk 2 above)
Comment (vacla) [reply src/main/java/net/coreprotect/command/CommandHandler.java]: Changed it to coreprotect:lookup

File: lang/en.yml (hunk 1 above)
Comment (vacla) [reply lang/en.yml:109]: Check the LookupCommand class, it doesn't fully show up since it has too many changes to display.

**Comment by Intelli** [comment]
I think a better way to approach this may be to create separate method(s) from the existing internal chat API. Modifying the [existing methods](https://github.com/PlayPro/CoreProtect/pull/211/files#diff-fb5e9a30f9606f690272a636ebd1cd5b1dad2adc8e189c5baa3c6b23906b2657) is a bit messy, and could create maintainability issues in the future. As well, things like [silencing chat](https://github.com/PlayPro/CoreProtect/pull/211/commits/75c0fd8acc5d4b0da25872550683c8eebe409ced#diff-a0792bef3911c24867485ff0f751d39af5d9030d5c967fd6d5501e977bf4f9b2) could then be a single simple switch within an existing chat method.

Things that can easily be implemented client-side should be left client-side, such as the [task that loops](https://github.com/PlayPro/CoreProtect/pull/211/files#diff-9dbeb2dae16faf0eb2618dcae165d2bd81fa513fa4b82b4cacafebb807dfb9be) through all the pages. The plugin already has server-side functionality for displaying multiple pages of data at once, namely the `rows:` and `/co lookup <page>:<rows>` parameters. The "Lookup Finished" phrase can also be removed, as it isn't something necessary to be displayed during lookups. 

To note, API changes or additions will no longer be pulled until before a major release. Only bug fixes will be pulled into the master branch prior to minor version releases.

**Comment by vacla** [comment]
@Intelli Would it be fine if i ran the runCommand internally inside CP instead of just letting the player run a command for the networking api?
And the reasons why i don't want to let the player sent the commands themself is:
1. Limited messages per seconds can be sent
2. Commands will be logged in the players chat history
(Copied from #development)

**Comment by Intelli** [comment]
> @Intelli Would it be fine if i ran the runCommand internally inside CP instead of just letting the player run a command for the networking api? And the reasons why i don't want to let the player sent the commands themself is:
> 
> 1. Limited messages per seconds can be sent
> 2. Commands will be logged in the players chat history
>    (Copied from #development)

1. CoreProtect already has internal command throttling. Triggering the commands internally would still have these throttles, so I doubt you'd see any real-world difference here.
2. CoreProtect is a logging plugin, so if a player runs a command, then it should be logged in that players command history.

**Pull request closed without merging**